v4/Models/Survey.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace v4\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Ushahidi\App\Repository\FormRepository;
+use Ushahidi\Core\Entity\Permission;
+use Ushahidi\Core\Tool\Permissions\InteractsWithFormPermissions;
+
+class Survey extends Model
+{
+    use InteractsWithFormPermissions;
+    protected $table = 'forms';
+    protected $with = ['stages'];
+    public $timestamps = FALSE;
+    /**
+     * The attributes that should be hidden for serialization.
+     * @note this should be changed so that we either use the fractal transformer
+     * OR a policy authorizer which is a more or less accepted method to do it
+     * (which uses the same $hidden type thing but it's much nicer obviously)
+     *
+     * @var array
+     */
+    protected $hidden = ['description'];
+    /**
+     * The attributes that should be mutated to dates.
+     * @var array
+    */
+    protected $dates = ['created', 'updated'];
+
+    /**
+    * The attributes that are mass assignable.
+    * @var array
+    */
+    protected $fillable = [
+        'parent_id',
+        'name',
+        'description',
+        'type',
+        'disabled',
+        'require_approval',
+        'everyone_can_create',
+        'color',
+        'hide_author',
+        'hide_time',
+        'hide_location',
+        'targeted_survey'
+    ];
+
+    /**
+     * The accessors to append to the model's array form.
+     *
+     * @var array
+     */
+    protected $appends = ['can_create'];
+    /**
+     * The model's default values for attributes.
+     *
+     * @var array
+     */
+    protected $attributes = [
+        'type' => 'report',
+        'require_approval' => true,
+        'everyone_can_create' => true,
+        'hide_author' => false,
+        'hide_time' => false,
+        'hide_location' => false,
+        'targeted_survey' => false
+    ];
+
+    /**
+     * This is what makes can_create possible
+     * @return mixed
+     */
+    public function getCanCreateAttribute() {
+        $can_create = $this->getCanCreateRoles($this->id);
+        return $can_create['roles'];
+    }
+//
+//    /**
+//     * This is what makes can_create possible
+//     * @return mixed
+//     */
+//    public function getTranslationsAttribute() {
+//        return $this->translations;
+//    }
+
+    private function getCanCreateRoles($form_id) {
+        /**
+         * @NOTE: to lower changes of a regression I'm using some helpers from
+         * repositories and traits we already have
+         * @NOTE: during origami and later stages of sunny buffers, we will fold this
+         * all together in more performant and friendly ways
+        */
+        $form_repo = service('repository.form');
+        return $form_repo->getRolesThatCanCreatePosts($form_id);
+    }
+    /**
+     * We check for relationship permissions here, to avoid hydrating anything that should not be hydrated.
+     * @return \Illuminate\Database\Eloquent\Relations\HasMany
+     */
+    public function stages()
```

Comment:
this is actually kind of nice? Not loving the usage of v3 authorizers but not much to do there. The good part is that we just do it in the hydration and it works seamlessly with the ->stages relation.